File: app/src/main/java/com/autilite/plan_g/program/BaseExercise.java
package com.autilite.plan_g.program;

import android.os.Parcel;

/**
 * Created by Kelvin on Aug 15, 2017.
 */

public class BaseExercise extends BaseModel {
    public BaseExercise(long id, String name, String description) {
        super(id, name, description);
    }

    public BaseExercise(Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<BaseExercise> CREATOR = new Creator<BaseExercise>() {
        @Override
        public BaseExercise createFromParcel(Parcel in) {
            return new BaseExercise(in);
        }

        @Override
        public BaseExercise[] newArray(int size) {
            return new BaseExercise[size];
        }
    };
}

File: app/src/main/java/com/autilite/plan_g/program/BaseExercise.kt
package com.autilite.plan_g.program

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Kelvin on Aug 15, 2017.
 */

class BaseExercise : BaseModel {
    constructor(id: Long, name: String, description: String) : super(id, name, description)

    protected constructor(parcel: Parcel) : super(parcel)

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<BaseExercise> {
        override fun createFromParcel(parcel: Parcel): BaseExercise {
            return BaseExercise(parcel)
        }

        override fun newArray(size: Int): Array<BaseExercise?> {
            return arrayOfNulls(size)
        }
    }
}
